app/controllers/scans_controller.rb

```diff
@@ -2,6 +2,13 @@
 #  Controller to handle particular behaviors for Scan type requests
 ###
 class ScansController < RequestsController
+  # Redirect to the Illiad url if we have a POST request
+  # OR if not a POST request then check if the user can create a Scan object
+  # (e.g. a GET request redirected from a webauth challenge)
+  before_action only: :create do
+    redirect_to illiad_url if request.post? || can?(:create, Scan)
```

Comment:
Switched this up to use a param that we pass in the URL we tell redirect Illiad back to us with.